File: healthhealthabc-api/src/main/java/com/chatchatabc/api/domain/service/UserService.java
package com.chatchatabc.api.domain.service;

import com.chatchatabc.api.application.dto.auth.AuthLoginRequest;
import com.chatchatabc.api.application.dto.auth.AuthRegisterRequest;
import com.chatchatabc.api.application.dto.user.UserDTO;

public interface UserService {

    String test(String value);

    /**
     * Register a new user
     */
    UserDTO register(AuthRegisterRequest userDTO, String roleName);

    /**
     * Login a user
     */
    String login(AuthLoginRequest authLoginRequest, String ipAddress);

    /**
     * Confirm registration
     */
    UserDTO confirmRegistration(String emailConfirmationId);

    /**
     * Generate recovery code for user to use
     */
    UserDTO forgotPassword(String email);

    /**
     * Reset password
     */
    UserDTO resetPassword(String email, String password, String recoveryCode);

    /**
     * Change password
     */
    void changePassword(String id, String oldPassword, String newPassword);

    /**
     * Update user profile
     */
    UserDTO updateProfile(String id, UserDTO userDTO);

    /**
     * Change email
     */
    UserDTO changeEmail(String id, String newEmail);

    /**
     * Confirm email change
     */
    UserDTO confirmEmailChange(String emailConfirmationId);
}

File: src/main/kotlin/com/chatchatabc/healthhealthabc/domain/service/UserService.kt
package com.chatchatabc.healthhealthabc.domain.service

import com.chatchatabc.healthhealthabc.domain.model.User
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
interface UserService : UserDetailsService {

    /**
     * Register a new user.
     */
    fun register(user: User, roleName: String = "ROLE_PATIENT"): User

    /**
     * Confirm a user's registration.
     */
    fun confirmRegistration(emailConfirmationId: String): User

    /**
     * Generate recovery code for user to use.
     */
    fun forgotPassword(email: String): User

    /**
     * Reset user's password.
     */
    fun resetPassword(email: String, password: String, recoveryCode: String): User

    /**
     * Change user's password.
     */
    fun changePassword(id: String, oldPassword: String, newPassword: String)

    /**
     * Update user's profile.
     */
    fun updateProfile(id: String, user: User): User
}